### DISCUSSION TITLE
fastapi  version==0.70.0 , backend reqeust will pending if browser refresh

### DISCUSSION THREAD

**Discussion by heyfavour** [fastapi/fastapi#8671] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class AccessMiddleware(BaseHTTPMiddleware):
    async def dispatch(self, request: Request, call_next):
        print("accept request")
        response = await call_next(request)
        print("finsh request")
        return response
```

### Description

i upgrade fastapi from 0.63.0 to 0.70.0,but i found reqeust will pending after browser refresh.
i use vue as frontend and  request the backend successfully ,then i wait 10 seconds ,F5 refresh brower，i found the fastapi get the reqeust but i will pending in " response = await call_next(request)  " (if i don't use BaseHTTPMiddleware ,it pending too)

### Operating System

Windows

### Operating System Details

windows 10 

### FastAPI Version

0.70.0

### Python Version

Python 3.9.1

### Additional Context

just print("accept request")

**Comment by STeveShary** [comment]
@heyfavour How is your FastAPI application running?  Are you using uvicorn?  Are you using gunicorn with uvicorn workers?

**Comment by heyfavour** [comment]
> @heyfavour How is your FastAPI application running? Are you using uvicorn? Are you using gunicorn with uvicorn workers?

just run            uvicorn.run(app='main:app', host="0.0.0.0", port=8080,reload=True)
i tried  upgrade uvicorn to lastest version ,and it does't work ,just downgrade fastapi works

**Comment by heyfavour** [comment]
> @heyfavour How is your FastAPI application running? Are you using uvicorn? Are you using gunicorn with uvicorn workers?

it seems like bug happends in fastapi/dependencies/utils.py->solve_dependencies  

# fastapi-routing-def get_request_handler()-def app()
![image](https://user-images.githubusercontent.com/21987565/140011128-054803ee-3f7a-4685-9bb6-ea1982e45ba3.png)
# fastapi/dependencies/utils.py->solve_dependencies  
![image](https://user-images.githubusercontent.com/21987565/140011140-244aa3af-bcda-4e0a-855c-0e0bcf156cd4.png)
# pycharm print
![image](https://user-images.githubusercontent.com/21987565/140011148-fe9250f0-d514-473e-98ee-19aff3078d00.png)

fastapi-routing-def get_request_handler()-def app()-await solve_dependencies 
if does run berfore solve_dependencies return 
but i does not print("fastapi routing.py solved_result end")


------------------------------------------------------------------------------------------------------
now i located in fastapi/dependencies/utils.py->func solve_generator

when execute func depency [get_db] it seems something happend in return await stack.enter_async_context(cm)
------------------------------------------------------------------------------------------------------
i found i seems like bugs in starlette, when i use starlette == 0.13.6 and fastapi ==0.70.0 it  works normlly.but fastapi==0.70.0 requires starlette==0.16.0

next is my bug new location
solve_generator->contextmanager_in_threadpool->run_in_threadpool(cm.__enter__)->anyio.to_thread.run_sync(func, *args)->get_asynclib().run_sync_in_worker_thread(func, *args, cancellable=cancellable,limiter=limiter)->anyio._backends._asyncio.run_sync_in_worker_thread:

it run until
    async with (limiter or current_default_thread_limiter()):
        with CancelScope(shield=not cancellable):
        ....
        try:
             return await future
       finally:
             .....

it pending in await future,but i don't know others code details so i don't konw why ,
i try recode :
    async with (limiter or current_default_thread_limiter()):
        with CancelScope(shield=not cancellable):
        ....
        try:
             print("start")
             result =  await future
             print("end")
             return result
       finally:
             .....
it just print("start"),so i think it pending in await future and something conflict in starlette.

next i don't konw how to debug

**Comment by STeveShary** [comment]
@heyfavour Hmmm... My guess is that you are not able to connect to your database.  At this point, I would recommend that you make a very small example of this.  If you still see the issue, then you can share it and post it here.  From the information provided, I'm not sure we can help you very much right now.

**Comment by heyfavour** [comment]
> @heyfavour Hmmm... My guess is that you are not able to connect to your database. At this point, I would recommend that you make a very small example of this. If you still see the issue, then you can share it and post it here. From the information provided, I'm not sure we can help you very much right now.

actually , i don't think that.i have try oracle and mysql ,it happend both two dbs. 


and i have try loacate in code which connect to db:
def get_db() -> Generator:
   print("this is a debut point") 
   try:
        db = SessionLocal()
        yield db
    finally:
        db.close()

it doesn't print!!!，so i don't think is a question about database connections.is seems like that it  doesn't execute in thread.

i am prefer to it's a confliction about  starlette's version 

next i will try another PC  to avoid only hapend in my PC.
